aggregator_core/src/datastore.rs

```diff
@@ -1310,61 +1312,77 @@ impl<C: Clock> Transaction<'_, C> {
             .collect::<Result<Vec<_>, Error>>()
     }
 
-    /// `get_unaggregated_client_report_ids_by_collect_for_task` returns pairs of report IDs and
+    /// `get_aggregatable_reports_for_time_interval_task` returns pairs of report IDs and
     /// aggregation parameters, corresponding to client reports that have not yet been aggregated,
     /// or not aggregated with a certain aggregation parameter, and for which there are collect
-    /// jobs, for a given task. Returned client reports are marked as aggregation-started, but this
-    /// will not stop additional aggregation jobs from being created later with different
-    /// aggregation parameters.
+    /// jobs, for a given time interval task. Returned client reports are marked as
+    /// aggregation-started, but this will not stop additional aggregation jobs from being created
+    /// later with different aggregation parameters.
     ///
     /// This should only be used with VDAFs with a non-unit type aggregation parameter. If a VDAF
     /// has the unit type as its aggregation parameter, then
     /// `get_unaggregated_client_report_ids_for_task` should be used instead. In such cases, it is
     /// not necessary to wait for a collection job to arrive before preparing reports.
     ///
-    /// This function deliberately ignores the `client_reports.aggregation_started` column, which
-    /// only has meaning for VDAFs without aggregation parameters.
+    /// Additionally, this method only works for time interval tasks, as it uses the collection
+    /// jobs' batch interval to determine if a report is included.
+    ///
+    /// This method updates the `aggregation_started` column, although it's not used to determine
+    /// whether a report is eligible for aggregation, since a report could be aggregated with
+    /// multiple aggregation parameters and thus a boolean does not suffice. Instead, that column is
+    /// used by [`Transaction::interval_has_unaggregated_reports`] to determine if a report has
+    /// never been aggregated.
+    // TODO(#225): It's possible we need to augment this so that the returned representation of an
```

Comment:
You're correct that it's only relevant to `interval_has_unaggregated_reports`. I think it's still useful to check that column in the multi-collected VDAF case, because it's a cheap way to account for the case of a report for which no aggregation job has ever been created. Otherwise we'd have to check whether any row for the report exists in `report_aggregations`, which would be much more expensive. I think avoiding that is why we have `client_reports.aggregation_started` in the first place.

Reply:
I do think we need to check `report_aggregations` (or do the same check more efficiently by the explicit report-to-batch association discussed in Slack).

In that case, checking `aggregation_started` as a fast path is probably not a win: it will only be useful on the very first aggregation. HH VDAFs like Poplar are likely to end up with many collections per batch; for all but the first, checking `aggregation_started` is an extra DB round-trip to no effect.

Reply:
FWIW, looking ahead to fixed size tasks with nontrivial aggregation parameters, we will also want to use the `aggregation_started` column to make assign reports to batches efficient.